### PR TITLE
Add primitive box :: OptionT a -> Possibly a

### PR DESCRIPTION
**Pull request by tranma**
now we can do

```
feature injury ~> days between (newest (box admitted)) now
```

rather than `feature injury ~> case admitted | None -> .... | Some a -> ...`

### CODE REVIEW

File: src/Icicle/Source/ToCore/Prim.hs [hunk 1]
```diff
@@ -212,6 +212,13 @@ convertPrim p ann resT xts = go p
    | otherwise
    = convertError
    $ ConvertErrorPrimNoArguments ann 2 p
+  -- Similarly this is not a tombstone.
+  godata Box
```
Comment (jystic) [inline src/Icicle/Source/ToCore/Prim.hs]: `godata` sounds like some scary golang keyword

Comment (tranma) [inline src/Icicle/Source/ToCore/Prim.hs]: below is `gomath` and `gomap`

**Comment by jystic** [comment]
Look good to me :+1:

File: src/Icicle/Source/ToCore/Prim.hs (hunk 1 above)
Comment (jystic) [inline src/Icicle/Source/ToCore/Prim.hs]: `gomath` sounds like you're gunna have a bad time

**Comment by amosr** [comment]
fantastic :facepunch:

**Pull request merged**